src/grt/src/GlobalRouter.cpp

```diff
@@ -2290,30 +2254,27 @@ void GlobalRouter::initGrid(int max_layer)
               generic_map);
 }
 
-void GlobalRouter::initRoutingLayers(std::vector<RoutingLayer>& routing_layers)
+void GlobalRouter::initRoutingLayers(std::map<int, odb::dbTechLayer*>& routing_layers)
 {
   odb::dbTech* tech = db_->getTech();
 
+  int valid_layers = 1;
   for (int l = 1; l <= tech->getRoutingLayerCount(); l++) {
     odb::dbTechLayer* tech_layer = tech->findRoutingLayer(l);
-    int index = l;
-    std::string name = tech_layer->getConstName();
-    bool preferred_direction;
-    if (tech_layer->getDirection().getValue()
-        == odb::dbTechLayerDir::HORIZONTAL) {
-      preferred_direction = RoutingLayer::HORIZONTAL;
-    } else if (tech_layer->getDirection().getValue()
-               == odb::dbTechLayerDir::VERTICAL) {
-      preferred_direction = RoutingLayer::VERTICAL;
-    } else {
-      logger_->error(GRT,
-                     84,
-                     "Layer {} does not have valid direction.",
-                     tech_layer->getName());
-    }
+    if (tech_layer->getLef58Type() != odb::dbTechLayer::MIMCAP) {
```

Comment:
Really getRoutingLayerCount & findRoutingLayer should handle this in the db as I don't think anybody wants to see mimcap as a routing layer.  Please move this to the db or let me know if you need help.

Reply:
Ok, this will make grt life easier. Besides all my updates to ignore the MIMCAP layers, I still have possible flaws related to pin layers, if they're placed at higher layers (like layer index 14 or something).